segment_coverage/segment_coverage/main.py

```diff
@@ -183,7 +188,13 @@ def run(self):
 					break
 
 				path = os.path.join(self.base_dir, self.channel, quality)
-				hours = [name for name in os.listdir(path) if not name.startswith('.')]
+				try:
+					hours = [name for name in os.listdir(path) if not name.startswith('.')]
+				except OSError as e:
+					if e.errno == errno.ENOENT:
+						self.logger.warning('{} does not exist'.format(path))
+						break
```

Comment:
I think this should be `continue` - even if one quality doesn't exist, others might. so we should try them all.